pages/bookmarks.jsx

```diff
@@ -3,54 +3,50 @@ import { getBookmarks } from '../js/bookmarks-manager';
 import ProjectList from '../components/project-list/project-list.jsx';
 import HintMessage from '../components/hint-message/hint-message.jsx';
 
-import UserData from '../js/user-data.js';
-import Login from '../js/login.js';
+import user from '../js/app-user.js';
 import env from "../config/env.generated.json";
 
 export default React.createClass({
   getInitialState() {
     return {
-      bookmarks: null
+      bookmarks: null,
+      user
     };
   },
+  handleSignInBtnClick(event) {
+    event.preventDefault();
+
+    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+    user.login(redirectUrl);
+  },
   handleLogOutBtnClick(event) {
     event.preventDefault();
-    Login.logoutUser(error => {
-      if (!error) {
-        this.forceUpdate();
-      }
-    });
+    user.logout();
+    this.setState({ user });
   },
   componentDidMount() {
     // get IDs of user's bookmarked entries
     this.setState({bookmarks: getBookmarks()});
 
     // let's verify what the logged in status is
-    let currentLocation = this.props.router.location;
-
-    Login.verifyLoggedInStatus(currentLocation, error => {
-      if (!error) {
-        this.forceUpdate();
-      }
-    });
+    user.verify(this.props.router.location, () => this.setState({ user }));
   },
   render() {
-    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
-    let logInUrl = UserData.logInUrl.get(redirectUrl);
-    let promptToLogIn = <p>Are you Mozilla Staff? <a href={logInUrl}>Sign in</a>.</p>;
-    let contentForStaff = <p>Hi {UserData.username.get()}! <button onClick={this.handleLogOutBtnClick} className="btn-link">Log out</button>.</p>;
-    let contentForNonStaff = <p>Only Mozilla staff can login now as we test this new platform. Meanwhile, your favorited projects can still be remembered on this device.</p>;
+    let anonymousContent = <p>Are you Mozilla Staff? <button className="btn btn-link" onClick={this.handleSignInBtnClick}>Sign in</button>.</p>;
+    let contentForLoggedInUser = <p>Hi {user.username}! <button onClick={this.handleLogOutBtnClick} className="btn-link">Log out</button>.</p>;
+    let failurePrompt = <p>Only Mozilla staff can login now as we test this new platform. Meanwhile, your favorited projects can still be remembered on this device.</p>;
+
+    let getloggedInStatus = function() {
```

Comment:
don't inline functions in `render`, just make them class functions and invoke them as `this.getLoggedInStatus()` etc. Especially in this case where `user` is global rather than local to the class instance.